### PR TITLE
Feature/3102023

### PR DESCRIPTION
**Pull request by bulgarchik**

### CODE REVIEW

File: FastBank.Services/CustomerService/CustomerService.cs [hunk 1]
```diff
@@ -40,37 +40,57 @@ public void Add(string name, string email, DateTime birthday, string password, R
...
-            //TODO validate age to be more then 18 and less then 100
+            CustomerAgeIsValid(customer, validationErrors);
             //TODO validate role
             return validationErrors;
         }
 
         public List<string> CustomerExist(Customer customer, List<string> validationErrors)
         {
-            var customers = _customerRepo.GetAll(); //TODO use IQueryable
+            var customers = _customerRepo.GetByEmail(customer.Email);
```
Comment (AngelovGrowth) [inline FastBank.Services/CustomerService/CustomerService.cs:59]: Customers??
If you expect only one customer, variable name should be - customer

**Review by AngelovGrowth — COMMENTED** [review]
Will be glad if you can resolve my comments

File: Infrastructure/DTOs/CustomerDTO.cs [hunk 2]
```diff
@@ -23,12 +24,13 @@ public CustomerDTO(Customer customer)
         public DateTime Birthday { get; private set; }
         public string Password { get; private set; }
         public string Role { get; private set; }
+        public bool InActive {  get; private set; }
```
[the remark this answers is not in the document]
Comment (bulgarchik) [reply Infrastructure/DTOs/CustomerDTO.cs:27]: No, An idea is to be false by default (inactive - false), only when we need to made user inactive we will set it to true.

File: Infrastructure/Context/FastBankDbContext.cs [hunk 3]
```diff
@@ -15,11 +15,11 @@ protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
...
         {
-            modelBuilder.Entity<CustomerDTO>().HasData(new CustomerDTO(new FastBank.Customer(Guid.NewGuid(), "Ivan", "1@1.com", DateTime.Now, "123", Roles.Customer)));
-            modelBuilder.Entity<CustomerDTO>().HasData(new CustomerDTO(new FastBank.Customer(Guid.NewGuid(), "Ангел Ангелов", "achoceo@abv.bg", DateTime.Now, "achkata", Roles.Manager)));
-            modelBuilder.Entity<CustomerDTO>().HasData(new CustomerDTO(new FastBank.Customer(Guid.NewGuid(), "Анелия Иванова", "ani90@abv.bg", DateTime.Now, "anito1990", Roles.Accountant)));
-            modelBuilder.Entity<CustomerDTO>().HasData(new CustomerDTO(new FastBank.Customer(Guid.NewGuid(), "Добромир Иванов", "dobaIv@abv.bg", DateTime.Now, "dobbanker", Roles.Banker)));
-            modelBuilder.Entity<CustomerDTO>().HasData(new CustomerDTO(new FastBank.Customer(Guid.NewGuid(), "Камелия Ангелова", "kamiang@abv.bg", DateTime.Now, "kameliq1988", Roles.CustomerService)));
+            modelBuilder.Entity<CustomerDTO>().HasData(new CustomerDTO(new FastBank.Customer(Guid.NewGuid(), "Ivan", "1@1.com", DateTime.Now, "123", Roles.Customer, false)));
```
[the remark this answers is not in the document]
Comment (bulgarchik) [reply Infrastructure/Context/FastBankDbContext.cs:18]: Maybe I should write it as inactive not inActive

File: FastBank.Services/CustomerService/CustomerService.cs (hunk 1 above)
Comment (bulgarchik) [reply FastBank.Services/CustomerService/CustomerService.cs]: I've modified it

File: FastBank.Domain/Customer.cs [hunk 4]
```diff
@@ -2,14 +2,15 @@
...
-        public Customer(Guid id, string name, string email, DateTime birthday, string password, Roles role)
+        public Customer(Guid id, string name, string email, DateTime birthday, string password, Roles role, bool inActive)
         {
             Id = id;
             Name = name;
             Email = email;
             Birthday = birthday;
             Password = password;
             Role = role;
+            Inactive = inActive;
```
Comment (AngelovGrowth) [inline FastBank.Domain/Customer.cs:13]: InActive = inActive

**Pull request merged**